### ISSUE TITLE
Received unexpected nonce with SDS after agent restart

### ISSUE THREAD

**Issue by mcpherrinm** [spiffe/spire#813]
After I restart/redeploy spire-agent, I receive a message that looks like this:

```
time="2019-03-22T00:38:17Z" level=warning msg="Received unexpected nonce \"bd60bdf7\" (expected \"\"); ignoring request" subsystem_name=sds_api
```

The source of that message is here:
https://github.com/spiffe/spire/blob/master/pkg/agent/endpoints/sds/handler.go#L102-L108

The comment there seems untrue, since the agent has been restarted.

**Comment by JonathanO** [comment]
On reconnecting to a discovery service Envoy will send a DiscoveryRequest with the last nonce from before it was disconnected. The newly restarted agent has no historical state, so doesn't recognise the nonce. Ignoring such a request leads to the reconnected Envoy never being sent updates.
Correct behaviour for the agent would be to only verify the nonce if the agent knows it has sent one.You can see this in the envoy go control plane example here: https://github.com/envoyproxy/go-control-plane/blob/0ad6fa1cf0b9b6ca8f3617a7188a568e81f40b87/pkg/server/server.go#L253

**Comment by azdagron** [comment]
@JonathanO fixed this in #820.
